tests/bsim/bluetooth/host/security/ccc_update/src/central.c

```diff
@@ -0,0 +1,391 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/bluetooth/att.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/hci_err.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+#include <zephyr/logging/log.h>
+
+#include <zephyr/settings/settings.h>
+
+#include "common.h"
+#include "settings.h"
+
+#include "argparse.h"
+#include "bs_pc_backchannel.h"
+
+#define CLIENT_CLIENT_CHAN 0
+#define SERVER_CLIENT_CHAN 1
+
+CREATE_FLAG(connected_flag);
+CREATE_FLAG(disconnected_flag);
+CREATE_FLAG(security_updated_flag);
+
+#define BT_UUID_DUMMY_SERVICE BT_UUID_DECLARE_128(DUMMY_SERVICE_TYPE)
+#define BT_UUID_DUMMY_SERVICE_NOTIFY BT_UUID_DECLARE_128(DUMMY_SERVICE_NOTIFY_TYPE)
+
+static struct bt_conn *default_conn;
+
+static struct bt_conn_cb central_cb;
+
+CREATE_FLAG(gatt_write_flag);
+static uint8_t gatt_write_att_err;
+
+static void gatt_write_cb(struct bt_conn *conn, uint8_t att_err,
+			  struct bt_gatt_write_params *params)
+{
+	gatt_write_att_err = att_err;
+
+	if (att_err) {
+		return;
+	}
+
+	SET_FLAG(gatt_write_flag);
+}
+
+static int gatt_write(struct bt_conn *conn, uint16_t handle, const uint8_t *write_buf,
+		      size_t write_size)
+{
+	int err;
+	struct bt_gatt_write_params params;
+
+	params.func = gatt_write_cb;
+	params.handle = handle;
+	params.offset = 0;
+	params.data = write_buf;
+	params.length = write_size;
+
+	UNSET_FLAG(gatt_write_flag);
+
+	/* `bt_gatt_write` is used instead of `bt_gatt_subscribe` and
+	 * `bt_gatt_unsubscribe` to bypass subscribtion checks of GATT client
+	 */
+	err = bt_gatt_write(conn, &params);
```

Comment:
Not sure I follow: Which checks are these? 

Doesn't the checks only exist on the server?

Reply:
The client is actually keeping track of his subscriptions in an array of `gatt_sub` structures. Doing a GATT write directly doesn't add anything to that structure.

The client does not care about the link state (encrypted or not), that's why, to make the test easier I don't use `bt_gatt_subscribe` not `bt_gatt_unsubscribe`.

Reply:
I am not sure I follow still. Looking at the implementation of `bt_gatt_subscribe` there are no checks for encryption state, so not sure why the link encryption state here matters at all.

Reply:
Yes, so we basically want to circumvent some of the behavior in the GATT client to act as a "bad client" in order to trigger some defensive/protective behavior on the server, right? 

Basically acting as a lower tester rather than a proper device

Reply:
Yes that's it.